components/properties/src/maps.rs

```diff
@@ -265,6 +265,12 @@ impl<'a, T: TrieValue> CodePointMapDataBorrowed<'a, T> {
             .filter(move |r| r.value == val)
             .map(|r| r.range)
     }
+
+    /// Yields an [`Iterator`] returning ranges of consecutive code points that
+    /// do *not* have the value `v` in the [`CodePointMapData`].
+    pub fn iter_ranges_for_value_complemented(self, val: T) -> impl Iterator<Item = RangeInclusive<u32>> + 'a {
```

Comment:
question: Should I expose this or a fully generic predicate API?

We can't do closures over FFI so there we will have to expose complement APIs anyway.